packages/wrangler/e2e/dev.test.ts

```diff
@@ -1087,6 +1087,110 @@ describe("watch mode", () => {
 					"<h1>Read more about Workers + Assets</h1>"
 				);
 			});
+
+			it(`supports switching from assets-only to Workers with assets during the current dev session`, async () => {
+				const helper = new WranglerE2ETestHelper();
+				await helper.seed({
+					"wrangler.toml": dedent`
+								name = "${workerName}"
+								compatibility_date = "2023-01-01"
+
+								[experimental_assets]
+								directory = "./public"
+						`,
+					"public/index.html": dedent`
+								<h1>Hello Workers + Assets</h1>`,
+				});
+				const worker = helper.runLongLived(cmd);
+				const { url } = await worker.waitForReady();
+
+				// verify response from Asset Worker
+				let { response } = await fetchWithETag(`${url}/index.html`, {});
```

Comment:
sorry the `fetchWithETag` helper is a bit clunky to use - if you don't return the cached ETags and pass it in the next time, it's not really doing anything.

like this:
```
let { response, cachedETags } = await fetchWithETag(`${url}/index.html`, {});
({ response, cachedETags} = await fetchWithETag(`${url}/index.html`, cachedETags));
// then you need should expect just a 304 not 200 with text etc.
expect(response.status).toBe(304)
```

Probably just use the regular fetch here since we're testing the eTag stuff separately.

Reply:
yeah, fair enough! I'll change the usage

Reply:
✅